File: app/src/main/java/com/cliniva/enventory/api/RetrofitClient.java
package com.cliniva.enventory.api;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitClient {

    private static Retrofit retrofit = null;

    public static RetrofitApi getInstance(){

        Gson gson = new GsonBuilder().setLenient().create();
        if (retrofit == null){
            String BASE_URL = "https://api.myjson.com/bins/";
            retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create(gson))
                    .build();
        }
        return retrofit.create(RetrofitApi.class);
    }
}

File: app/src/main/java/com/amirsons/inventory/api/RetrofitClient.kt
package com.amirsons.inventory.api

import com.google.gson.GsonBuilder
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitClient {

    const val BASE_URL = "https://api.myjson.com/bins/"
    private var api: RetrofitApi? = null

    val instance: RetrofitApi
        get() {

            if (api == null) {

                val gson = GsonBuilder().setLenient().create()

                val retrofit = Retrofit.Builder()
                        .baseUrl(BASE_URL)
                        .addConverterFactory(GsonConverterFactory.create(gson))
                        .build()

                api = retrofit.create(RetrofitApi::class.java)
            }

            return api!!
        }
}
